GoogleChromeLabs/ps-analysis-tool
# Feat: Export cookie table data from the extension

## Description


Adds an export button to the table component which allows the user to export the data shown in a table as a CSV file

## Relevant Technical Choices
- The above functionality was added to the table component itself for the export capability to be reusable.
- If any filters are applied the exported CSV file will have the filtered output.
- Only the CSV blob generation utility for cookie data is added in `packages/design-system/src/components/table/utils/generateCookieTableCSV.ts` for other types which can be shown in the table component new utilities will be added later in the same directory



## Testing Instructions
- Select any frame with some cookies.
- The export button in the table top bar should be able to instantiate a download for a file named "out.csv" which will have the data about the cookies in the table.
- If any filters are applied, the file "out.csv" should only have cookies that pass those filters.



## Screenshot
![Screenshot 2024-01-29 at 12 16 27 PM](https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/53055971/37bc7e7e-0a60-432e-b5b8-c8ff2a01aafd)




---

## Checklist






Fixes #275
